ConstructorIcon.svelte

```diff
@@ -0,0 +1,236 @@
+
+            swatches: [
+                'rgba(244, 67, 54, 1)',
+                'rgba(233, 30, 99, 0.95)',
+                'rgba(156, 39, 176, 0.9)',
+                'rgba(103, 58, 183, 0.85)',
+                'rgba(63, 81, 181, 0.8)',
+                'rgba(33, 150, 243, 0.75)',
+                'rgba(3, 169, 244, 0.7)',
+                'rgba(0, 188, 212, 0.7)',
+                'rgba(0, 150, 136, 0.75)',
+                'rgba(76, 175, 80, 0.8)',
+                'rgba(139, 195, 74, 0.85)',
+                'rgba(205, 220, 57, 0.9)',
+                'rgba(255, 235, 59, 0.95)',
+                'rgba(255, 193, 7, 1)'
+            ],
+
+            components: {
+                // Main components
+                preview: true,
+                opacity: true,
+                hue: true,
+
+                // Input / output Options
+                interaction: {
+                    hex: true,
+                    input: true,
+                    clear: true,
+                    save: true
+                }
+            }
+        });
+
+        pickr.on('save', (color, instance) => {
+            console.log('Event: "save"', color, instance);
+            colour = color.toHEXA().toString();
+            console.log("Colour is now", colour);
+        });
+    }
+
+
+    onMount(() => {
+        createPickr(picker)
+        // const pickr = Pickr.create({
+        //     el: picker,
+        //     theme: 'nano', // or 'monolith', or 'nano'
+        //     container: container,
+
+        //     swatches: [
+        //         'rgba(244, 67, 54, 1)',
+        //         'rgba(233, 30, 99, 0.95)',
+        //         'rgba(156, 39, 176, 0.9)',
+        //         'rgba(103, 58, 183, 0.85)',
+        //         'rgba(63, 81, 181, 0.8)',
+        //         'rgba(33, 150, 243, 0.75)',
+        //         'rgba(3, 169, 244, 0.7)',
+        //         'rgba(0, 188, 212, 0.7)',
+        //         'rgba(0, 150, 136, 0.75)',
+        //         'rgba(76, 175, 80, 0.8)',
+        //         'rgba(139, 195, 74, 0.85)',
+        //         'rgba(205, 220, 57, 0.9)',
+        //         'rgba(255, 235, 59, 0.95)',
+        //         'rgba(255, 193, 7, 1)'
+        //     ],
+
+        //     components: {
+
+        //         // Main components
+        //         preview: true,
+        //         opacity: true,
+        //         hue: true,
+
+        //         // Input / output Options
+        //         interaction: {
+        //             hex: true,
+        //             input: true,
+        //             clear: true,
+        //             save: true
+        //         }
+        //     }
+        // });
+
+        // pickr.on('save', (color, instance) => {
+        //     console.log('Event: "save"', color, instance);
+        //     colour = color.toHEXA().toString();
+        //     console.log("Colour is now", colour);
+        // });
+    });
+    export let commitExpression
+
+    let expData = {
```

Comment:
I'm not sure how is this struct can be reused by another concrete DNA like group-expression.